### ISSUE TITLE
ERROR unexpected EOF

### ISSUE THREAD

**Issue by IChocked** [suyashkumar/dicom#109]
### Version
Pre-release v1.0.0-alpha.2

### Background
Setup a simple test to create a `NewParser` and parse ~2 million DICOMs files. Almost all of them were fine and read correctly without issue, but on 404 of them, one of the following two errors was given:
`ERROR unexpected EOF`, or related, `ERROR error reading tag: unexpected EOF EOF`

I have not yet had a chance to investigate where this error comes from or what might be causing it, but below I've attached some example files for each error to help reproduce.

### Implementation of the Library
The following is the main/relevant section of what was iterated over all the sample DICOMs.
```
file, err := os.Open(location)
        if err != nil {
                return nil, err
        }

        info, err := file.Stat()
        if err != nil {
                return nil, err
        }

        p, err := dicom.NewParser(file, info.Size(), nil)
        if err != nil {
                return nil, err
        }

        parsedData, err := p.Parse()
        if parsedData.Elements == nil || err != nil {
                return nil, err
        }
```

### Sample DICOMs
[unexpected_eof.zip](https://github.com/suyashkumar/dicom/files/5043308/unexpected_eof.zip)
[error_reading_tag.zip](https://github.com/suyashkumar/dicom/files/5043309/error_reading_tag.zip)

### Credits
- _Work on this bug report was sponsored by: [Segmed Inc.](https://www.segmed.ai)_

- _Fenimore, Charles, McNitt-Gray, Michael F., Clunie, David, Gavrielides, Marios A., Petrick, Nicholas, Samei, Ehsan, … Slazak, Katherine. (2016). Data from QIBA CT-1C. The Cancer Imaging Archive. DOI: 10.7937/K9/TCIA.2016.YxgR4blU_

- _Clark K, Vendt B, Smith K, Freymann J, Kirby J, Koppel P, Moore S, Phillips S, Maffitt D, Pringle M, Tarbox L, Prior F. (2013) The Cancer Imaging Archive (TCIA): Maintaining and Operating a Public Information Repository, Journal of Digital Imaging, Volume 26, Number 6 pp 1045-1057. DOI: 10.1007/s10278-013-9622-7_

**Comment by suyashkumar** [comment]
@IChocked just wanted to say another big thank you to you and the folks at Segmed for helping with some of the library validation for the rewritten release!

Thank you for identifying this issue and documenting it so well. I was able to put together a fix in #110, and will add samples from the provided DICOMs to a test set for the rewritten library in an upcoming pull request :). 

Step by step we're getting closer to the rewritten release, can't wait!

**Comment by suyashkumar** [comment]
Also, tagged the newest alpha as `v1.0.0-alpha.3` that includes this fix (and several other changes)!
